backend/src/services/langchain_service.py

```diff
@@ -0,0 +1,16 @@
+import os
+from langchain.llms import OpenAI
+from langchain import LLMChain, PromptTemplate
+
+hint_prompt_template = '''...'''
+hint_prompt = PromptTemplate.from_template(template=hint_prompt_template)
+
+def get_llm() -> OpenAI:
+    openai_api_key = os.getenv("OPENAI_API_KEY")
+    return OpenAI(openai_api_key=openai_api_key, temperature=0.5, max_tokens=1024)
+    
+def get_hint_chain(description: str, code: str) -> str:
+    llm = get_llm()
+    hint_chain = LLMChain(llm=llm, prompt=hint_prompt)
```

Comment:
This is meaningless at the moment, but do you have to call OpenAI() and LLMChain() for every request? I assume OpenAI forms the connection and LLMChain just sets up the ability to chain prompts together with the initial prompt being `hint_prompt`.

Reply:
I have a similar question - is there a way we can establish that connection to OpenAI when the server first starts up? Then keep re-using that connection for all hint requests

Reply:
Yeah we can attach it to the fastAPI app instance.

Reply:
I think we can do that when we migrate the LLM fully to llama in a separate PR